lib/diaspora_federation/salmon/xml_payload.rb

```diff
@@ -85,19 +93,42 @@ def self.entity_class_name(term)
       # @param [Nokogiri::XML::Element] node xml nodes
       # @return [Entity] instance
       def self.populate_entity(klass, node)
-        data = {}
-        klass.class_props.each do |prop_def|
-          name = prop_def[:name]
+        # Build an auxillary hash to check whether XML node is a defined member of klass
+        props = klass.class_props.map { |prop_def|
           type = prop_def[:type]
-
           if type == String
-            data[name] = parse_string_from_node(prop_def[:xml_name], node)
+            [prop_def[:xml_name], prop_def]
           elsif type.instance_of?(Array)
-            data[name] = parse_array_from_node(type, node)
+            [type.first.entity_name.to_sym, prop_def]
```

Comment:
I haven't looked close to ```PropertiesDSL``` but I also think ```populate_entity``` is too complex and it's worth trying to find a way to simplify it.